syntax = "proto2";

import "google/protobuf/any.proto";

package dpe;

message ConnectRequest {
  optional string address = 1;
}

message ConnectResponse {
  optional int32 connection_id = 1;
}

message DisconnectRequest {
  optional string address = 1;
}

message ComputeRequest {
  optional int32 task_id = 1;
}

message FinishComputeRequest {
  optional int32 task_id = 1;
  optional string result = 2;
}


message Request {
  optional string name = 1;
  optional int64 request_id = 2 [default = 0];
  optional int32 connection_id = 3 [default = 0];

  optional int64 timestamp = 100 [default = 0];
  
  optional ConnectRequest connect = 300;
  optional DisconnectRequest disconnect = 301;
  optional ComputeRequest compute = 302;
  optional FinishComputeRequest finish_compute = 303;
}

message Response {
  optional string name = 1;
  optional int64 error_code = 2;

  optional int64 timestamp = 100 [default = 0];
  
  optional int64 request_timestamp = 200 [default = 0];
  optional int64 request_id = 201 [default = 0];
  
  optional ConnectResponse connect = 300;
}


